Rakefile

```diff
@@ -19,27 +36,30 @@ end
 
 namespace :docs do
   task :prepare do
-    `cp yard-config ~/.yard/config`
     Dir.mkdir("doc")
     Dir.chdir("doc") do
-      `git init`
-      `git remote add origin git@github.com:bugsnag/maze-runner.git`
-      `git fetch`
-      `git checkout gh-pages`
-      `git pull`
+      run_command "git init"
+      run_command "git remote add origin https://github.com/bugsnag/maze-runner.git"
+      run_command "git fetch"
+      run_command "git checkout gh-pages"
+      run_command "git pull"
+      run_command "echo https://#{ENV["DOCS_PUSH_TOKEN"]}:x-oauth-basic@github.com >> ~/.git-credentials"
+      run_command "git config credential.helper 'store'"
+      run_command "git config --global user.email 'notifiers@bugsnag.com'"
+      run_command "git config --global user.name 'Bugsnag notifiers'"
```

Comment:
Can this go in the Dockerfile.docs?

Reply:
It can, only potential issue would be the push token being uploaded in plain-text to our AWS ECR repository.  Probably not a big deal, but it's a potential risk.

Reply:
As a note, AWS S3 already contains the token in plain text which is probably an equivalent risk, so this is likely a non-issue.

Reply:
I don't have a strong view either way, it just seemed like a one-time setup step rather than part of the publish steps.

Reply:
Realised this is a non issue as this image isn't uploaded to ECR.  The token doesn't leave CI.